engine/crates/engine/src/schema.rs

```diff
@@ -186,8 +172,10 @@ impl Deref for SchemaEnv {
 #[doc(hidden)]
 pub struct SchemaInner {
     pub(crate) validation_mode: ValidationMode,
-    pub(crate) complexity: Option<usize>,
-    pub(crate) depth: Option<usize>,
+    pub(crate) max_complexity: Option<usize>,
```

Comment:
Could also be in a `OperationLimits` struct with a `Default`

Reply:
Good idea!